ktor-utils/common/src/io/ktor/util/ByteChannels.kt

```diff
@@ -42,6 +42,33 @@ fun ByteReadChannel.split(coroutineScope: CoroutineScope): Pair<ByteReadChannel,
     return first to second
 }
 
+/**
+ * Copy source channel to both output channels chunk by chunk.
+ */
+@InternalAPI
+fun ByteReadChannel.copyToBoth(first: ByteWriteChannel, second: ByteWriteChannel) {
+    GlobalScope.launch(Dispatchers.Unconfined) {
+        try {
+            while (!isClosedForRead && (!first.isClosedForWrite || !second.isClosedForWrite)) {
+                readRemaining(CHUNK_BUFFER_SIZE).use {
+                    try {
+                        first.writePacket(it.copy())
+                    } catch (_: Throwable) {
```

Comment:
Do we intentionally miss all exceptions here?

Reply:
Yep, it's only for logging purposes